lib/package_utils.rb

```diff
@@ -41,6 +48,32 @@ def self.get_sha256(pkg, build_from_source: false)
     end
   end
 
+  def self.get_binary_url(pkg)
+    # List all the packages with the name and version of the package file.
+    # The name search is fuzzy, so we need to refine it further (otherwise packages like vim, gvim and vim_runtime would break).
+    packages = JSON.parse(Net::HTTP.get(URI("https://gitlab.com/api/v4/projects/26210301/packages?package_name=#{pkg.name}&package_version=#{pkg.version}_#{ARCH}")))
+    # Loop over each result until we get an exact name match, then return the package ID for that match.
+    package_id = 0
+    (0..packages.count - 1).each do |i|
+      next unless packages[i]['name'] == pkg.name
+      package_id = packages[i]['id']
+    end
+    # Return early if we weren't able to find the package ID, so that the CREW_CACHE_ENABLED hack to test packages without uploading them still works.
+    # When we're doing that, we're calling download knowing that there isn't an actual file to download, but relying on CREW_CACHE_ENABLED to save us before we get there.
+    return if package_id.zero?
+    # List all the package files for that package ID.
+    package_files = JSON.parse(Net::HTTP.get(URI("https://gitlab.com/api/v4/projects/26210301/packages/#{package_id}/package_files")))
+    # Bail out if we weren't actually able to find a package.
+    return if package_files.is_a?(Hash) && package_files['message'] == '404 Not found'
+    # Loop over each result until we find a matching file_sha256 to our binary_sha256.
+    (0..package_files.count - 1).each do |i|
+      next unless package_files[i]['file_sha256'] == pkg.binary_sha256[ARCH.to_sym]
+      return "https://gitlab.com/chromebrew/binaries/-/package_files/#{package_files[i]['id']}/download"
+    end
+    # If we're still here, the likely cause is that the file sha256s are mismatched. Let the user figure it out.
+    return
```

Comment:
Providing a return value helps with debugging.  If you return false then you can make decisions from the calling function.

Reply:
I'm using an empty return here rather than return false for brevity (we have 3 possible early return places, so it's easier to specify `if possible_url.to_s.empty?` in the calling function rather than saying `return false` three times in order to only have to write `if possible_url` in the calling function.

Reply:
I don't think you understand what I'm trying to suggest here.  If this function returns nothing and you "let the user figure it out", they would have no idea why this function failed to get a url.  If you returned an error code with a message, on the other hand, that would be much more helpful to figure out what is wrong.

Reply:
Oh, that's a documentation error on my part. What I mean is that this situation will be handled by the calling function.

Reply:
```
return fallback_url
```